soh/soh/ActorDB.cpp

```diff
@@ -0,0 +1,617 @@
+    newEntry.SetDesc(desc);
+
+    newEntry.entry.valid = true;
+
+    return newEntry;
+}
+
+// Adds an actor with the legacy ActorInit struct. The id is provided in it.
+ActorDB::Entry& ActorDB::AddEntry(const std::string& name, const std::string& desc, const ActorInit& init) {
+    Entry& entry = AddEntry(name, desc, init.id);
+
+    entry.entry.category = init.category;
+    entry.entry.flags = init.flags;
+    entry.entry.objectId = init.objectId;
+    entry.entry.instanceSize = init.instanceSize;
+    entry.entry.init = init.init;
+    entry.entry.destroy = init.destroy;
+    entry.entry.update = init.update;
+    entry.entry.draw = init.draw;
+    entry.entry.reset = init.reset;
+
+    return entry;
+}
+
+// Adds an actor with the new ActorDBInit struct. The id assigned to the actor is dynamic. Use the return Entry or RetrieveId to get it.
+ActorDB::Entry& ActorDB::AddEntry(const ActorDBInit& init) {
+    Entry& entry = AddEntry(init.name, init.desc, nextFreeId);
+
+    entry.entry.category = init.category;
+    entry.entry.flags = init.flags;
+    entry.entry.objectId = init.objectId;
+    entry.entry.instanceSize = init.instanceSize;
+    entry.entry.init = init.init;
+    entry.entry.destroy = init.destroy;
+    entry.entry.update = init.update;
+    entry.entry.draw = init.draw;
+    entry.entry.reset = init.reset;
+
+    return entry;
+}
+
+// Get the ActorDB::Entry for the given actor id.
+ActorDB::Entry& ActorDB::RetrieveEntry(const int id) {
+    static Entry invalid;
+    if ((id < 0) || (id >= db.size())) {
+        return invalid;
+    }
+    return db[id];
+}
+
+// Get the id for a given actor by name.
+int ActorDB::RetrieveId(const std::string& name) {
+    auto entry = nameTable.find(name);
+    if (entry == nameTable.end()) {
+        return -1;
+    }
+
+    return entry->second;
+}
+
+ActorDB::Entry::Entry() {
+    entry.name = nullptr;
+    entry.desc = nullptr;
+    entry.valid = false;
+    entry.id = 0;
+    entry.category = 0;
+    entry.flags = 0;
+    entry.objectId = 0;
+    entry.instanceSize = 0;
+    entry.init = nullptr;
+    entry.destroy = nullptr;
+    entry.update = nullptr;
+    entry.draw = nullptr;
+    entry.reset = nullptr;
+    entry.numLoaded = 0;
+}
+
+ActorDB::Entry::Entry(const Entry& other) {
+    entry = other.entry;
+    SetName(other.name);
+    SetDesc(other.desc);
+}
+
+ActorDB::Entry& ActorDB::Entry::operator=(const Entry& other) {
+    entry = other.entry;
+    SetName(other.name);
+    SetDesc(other.desc);
+    return *this;
+}
+
+void ActorDB::Entry::SetName(const std::string& newName) {
+    name = newName;
+    entry.name = name.c_str();
+}
+
+void ActorDB::Entry::SetDesc(const std::string& newDesc) {
+    desc = newDesc;
+    entry.desc = desc.c_str();
+}
+
+#include "src/overlays/actors/ovl_En_Partner/z_en_partner.h"
+static ActorDBInit EnPartnerInit = {
+    "En_Partner",
+    "Ivan",
+    ACTORCAT_ITEMACTION,
+    (ACTOR_FLAG_4 | ACTOR_FLAG_5 | ACTOR_FLAG_10 | ACTOR_FLAG_26),
```

Comment:
Actor flags were renamed/doc-ed in a recent PR. Need these changes to compile.
```suggestion
    (ACTOR_FLAG_UPDATE_WHILE_CULLED | ACTOR_FLAG_DRAW_WHILE_CULLED | ACTOR_FLAG_DRAGGED_BY_HOOKSHOT | ACTOR_FLAG_CAN_PRESS_SWITCH),
```

Reply:
Thanks for the fix!